joint_state_publisher/src/joint_state_publisher/__init__.py

```diff
@@ -134,6 +112,15 @@ def init_urdf(self, robot):
     def __init__(self):
         description = get_param('robot_description')
 
+        use_gui = get_param('use_gui')
+        if use_gui is not None:
+            rospy.logerr('GUI is no longer supported by this package; install and run joint_state_publisher_gui')
+            sys.exit(1)
```

Comment:
I strongly believe we should approach this the same way we're working on removing the GUI dependencies from `actionlib` in ros/actionlib#157.

That is, rather than failing outright, attempt to minimize the breaking change by searching for and launching `joint_state_publisher_gui` if possible, with a warning telling users about the change. In the case that the `joint_state_publisher_gui` package isn't found, print an error and exit(1) as you've done here.